### ISSUE TITLE
Notifications buggy with HPA trigger

### ISSUE THREAD

**Issue by deepanverma19** [argoproj/argo-rollouts#2011]
## Summary 

What happened/what you expected to happen?
Whenever a HPA is triggered, the number of pods increase which is the expected behavior but when the downscaling happens i.e. for example pods go from 3 to 2, the rollout gets degraded and sends red alert notification that it has been degraded and after its back to original 2/2, should also be sending green alert that rollout is stable whereas it doesn't send (only red alerts). In short, the rollout shouldn't be triggering notifications on events rollout-completed or rollout-degraded (customized trigger) whenever HPA does downscaling.

## Diagnostics

What version of Argo Rollouts are you running?
v1.1.0
```
# Using the configmap as below


  template.rollout-completed: |
    message: Rollout {{"{{.rollout.metadata.name}}"}} has been completed.
    email:
      subject: Rollout {{"{{.rollout.metadata.name}}"}} has been completed.
    slack:
      attachments: |
          [{
            "title": "{{"{{ .rollout.metadata.name}}"}}",
            "color": "#18be52",
            "fields": [
            {
              "title": "Strategy",
              "value": "{{"{{if .rollout.spec.strategy.blueGreen}}"}}BlueGreen{{"{{end}}"}}{{"{{if .rollout.spec.strategy.canary}}"}}Canary{{"{{end}}"}}",
              "short": true
            }
            {{"{{range $index, $c := .rollout.spec.template.spec.containers}}"}}
              {{"{{if not $index}}"}},{{"{{end}}"}}
              {{"{{if $index}}"}},{{"{{end}}"}}
              {
                "title": "{{"{{$c.name}}"}}",
                "value": "{{"{{$c.image}}"}}",
                "short": true
              }
            {{"{{end}}"}}
            ]
          }]
  template.rollout-degraded: |
    message: Rollout {{"{{.rollout.metadata.name}}"}} has been degraded.
    email:
      subject: Rollout {{"{{.rollout.metadata.name}}"}} has been degraded.
    slack:
      attachments: |
          [{
            "title": "{{"{{ .rollout.metadata.name}}"}}",
            "color": "#be1818",
            "fields": [
            {
              "title": "Strategy",
              "value": "{{"{{if .rollout.spec.strategy.blueGreen}}"}}BlueGreen{{"{{end}}"}}{{"{{if .rollout.spec.strategy.canary}}"}}Canary{{"{{end}}"}}",
              "short": true
            }
            {{"{{range $index, $c := .rollout.spec.template.spec.containers}}"}}
              {{"{{if not $index}}"}},{{"{{end}}"}}
              {{"{{if $index}}"}},{{"{{end}}"}}
              {
                "title": "{{"{{$c.name}}"}}",
                "value": "{{"{{$c.image}}"}}",
                "short": true
              }
            {{"{{end}}"}}
            ]
          }]
  trigger.on-rollout-completed: |
    - send: [rollout-completed]
  trigger.on-rollout-degraded: |
    - send: [rollout-degraded]
      when: rollout.status.phase == 'Degraded'
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by harikrongali** [comment]
@RaviHari can you please check?

**Comment by RaviHari** [comment]
@deepanverma19 I tried using HPA and whenever scaleup / scaledown happens the rollout status phase is healthy it is not degraded. The events generated are on ScalingReplicaSet not on rollout-completed / rollout-degraded.

```
Status:
  HPA Replicas:        4
  Available Replicas:  4
  Blue Green:
  Canary:
  Conditions:
    Last Transition Time:  2022-06-30T10:17:46Z
    Last Update Time:      2022-06-30T10:17:46Z
    Message:               Rollout is paused
    Reason:                RolloutPaused
    Status:                False
    Type:                  Paused
    Last Transition Time:  2022-06-30T10:49:49Z
    Last Update Time:      2022-06-30T10:49:49Z
    Message:               Rollout has minimum availability
    Reason:                AvailableReason
    Status:                True
    Type:                  Available
    Last Transition Time:  2022-06-30T10:56:55Z
    Last Update Time:      2022-06-30T10:56:55Z
    Message:               RolloutCompleted
    Reason:                RolloutCompleted
    Status:                True
    Type:                  Completed
    Last Transition Time:  2022-06-30T10:17:46Z
    Last Update Time:      2022-06-30T10:56:55Z
    Message:               ReplicaSet "rollouts-demo-6c87dbb745" has successfully progressed.
    Reason:                NewReplicaSetAvailable
    Status:                True
    Type:                  Progressing
  Current Pod Hash:        6c87dbb745
  Current Step Hash:       6f78cbd569
  Current Step Index:      7
  Observed Generation:     17
  Phase:                   Healthy
  Ready Replicas:          4
  Replicas:                4
  Selector:                app=rollouts-demo
  Stable RS:               6c87dbb745
  Updated Replicas:        4
Events:
  Type    Reason                Age   From                 Message
  ----    ------                ----  ----                 -------
  Normal  RolloutUpdated        56m   rollouts-controller  Rollout updated to revision 3
  Normal  NewReplicaSetCreated  56m   rollouts-controller  Created ReplicaSet rollouts-demo-7b64f8bd6f (revision 3)
  Normal  ScalingReplicaSet     24m   rollouts-controller  Scaled up ReplicaSet rollouts-demo-6c87dbb745 (revision 4) from 4 to 5
  Normal  ScalingReplicaSet     24m   rollouts-controller  Scaled up ReplicaSet rollouts-demo-6c87dbb745 (revision 4) from 5 to 7
  Normal  ScalingReplicaSet     24m   rollouts-controller  Scaled up ReplicaSet rollouts-demo-6c87dbb745 (revision 4) from 7 to 10
  Normal  ScalingReplicaSet     18m   rollouts-controller  Scaled down ReplicaSet rollouts-demo-6c87dbb745 (revision 4) from 10 to 9
  Normal  ScalingReplicaSet     18m   rollouts-controller  Scaled down ReplicaSet rollouts-demo-6c87dbb745 (revision 4) from 9 to 8
  Normal  ScalingReplicaSet     17m   rollouts-controller  Scaled down ReplicaSet rollouts-demo-6c87dbb745 (revision 4) from 8 to 7
  Normal  ScalingReplicaSet     17m   rollouts-controller  Scaled down ReplicaSet rollouts-demo-6c87dbb745 (revision 4) from 7 to 6
  Normal  ScalingReplicaSet     17m   rollouts-controller  Scaled down ReplicaSet rollouts-demo-6c87dbb745 (revision 4) from 6 to 5
```

Can you please re-verify your use case.

**Comment by deepanverma19** [comment]
@RaviHari my best guess here is probably it works fine for normal HPA but is having issues with [keda](https://keda.sh/) which we're using. I will close this issue for now and reopen one later (if required), thanks for looking into this! Also, I see `rollout-aborted` was added recently to the config, so I replaced our customized `rollout-degraded` with `rollout-aborted` Thanks, that was helpful.
